packages/foam-core/src/utils/log.ts

```diff
@@ -0,0 +1,89 @@
+export interface ILogger {
+  debug(message?: any, ...params: any[]): void;
+  info(message?: any, ...params: any[]): void;
+  warn(message?: any, ...params: any[]): void;
+  error(message?: any, ...params: any[]): void;
+  getLevel(): LogLevelThreshold;
+  setLevel(level: LogLevelThreshold): void;
+}
+
+export type LogLevel = 'debug' | 'info' | 'warn' | 'error';
```

Comment:
I think this makes more sense as an `enum`, as well as `LogLevelThreshold` and `severity`

Reply:
it started as an enum, the reason why I turned it into a union type is because as I was using them, having enums was making the code clunkier.
Per https://stackoverflow.com/questions/40275832/typescript-has-unions-so-are-enums-redundant for the usage we make of log levels/thresholds, the two seemed equivalent to me, but happy to reconsider.

What's your thinking?

Reply:
The strings as types approach seems error prone while `Enum.Something` seems better.
If the types however are just compile-time this approach is better